### PR TITLE
fix potential threading issue in hetero when getting blob of same infer request in parallel

### PR DESCRIPTION
**Pull request by songbell**
Signed-off-by: fishbell <bell.song@intel.com>

### Details:
 - resolve comments related with hetero in PR [https://github.com/openvinotoolkit/openvino/pull/11934] and PR [https://github.com/openvinotoolkit/openvino/pull/11590]

### Tickets:
 - CVS-94326

### CODE REVIEW

File: src/plugins/hetero/async_infer_request.cpp [hunk 1]
```diff
@@ -59,6 +59,8 @@ StatusCode HeteroAsyncInferRequest::Wait(int64_t millis_timeout) {
 
 InferenceEngine::Blob::Ptr HeteroAsyncInferRequest::GetBlob(const std::string& name) {
     CheckState();
```
Comment (songbell) [inline src/plugins/hetero/async_infer_request.cpp:61]: test

**Comment by songbell** [comment]
@ilya-lavrenov  ilya, please help review

File: src/plugins/hetero/async_infer_request.cpp [hunk 2]
```diff
@@ -59,6 +59,8 @@ StatusCode HeteroAsyncInferRequest::Wait(int64_t millis_timeout) {
...
+    // need lock when different blobs of same infer request from parallel
+    std::lock_guard<std::mutex> lock(_getBlobMutex);
```
Comment (ilyachur) [inline src/plugins/hetero/async_infer_request.cpp:63]: Why do we need to have this mutex? Does line `setPointerToSo(_heteroInferRequest->getPointerToSo());` cause the issue?

**Review by ilyachur — COMMENTED** [review]
Could you add new tests?

File: src/plugins/hetero/async_infer_request.cpp (hunk 2 above)
Comment (songbell) [reply src/plugins/hetero/async_infer_request.cpp:63]: yes, the setPointerToSo will have race condition, when getting different blobs of same infer request in hetero, as in hetero, these blobs may come from different hw. I'll check and add the test cases

Comment (ilya-lavrenov) [reply src/plugins/hetero/async_infer_request.cpp:63]: Can we remove global setsopointer? In this case we don't need any mutexes

**Comment by ilya-lavrenov** [comment]
Please, add tests

**Comment by akladiev** [comment]
This PR will be closed in 2 weeks in case of no activity.

**Pull request closed without merging**